docs/docs/user/references/components.md

```diff
@@ -0,0 +1,245 @@
+# Components
+
+This section explains the different components of KPOps, their usage and
+configuration. To learn more about their hierarchy, visit the
+[architecture](/docs/developer/architecture/component-inheritance.md) page.
+
+## KubernetesApp
+
+### Usage
+
+Can be used to deploy any app in Kubernetes using Helm, for example, a REST
+service that serves Kafka data.
+
+### Configuration
+
+??? "`pipeline.yaml`"
+
+    ```yaml
+      --8<--
+      ./docs/resources/pipeline-components/kubernetes-app.yaml
+      --8<--
+    ```
+
+### Operations
+
+#### deploy
+
+Deploy using Helm.
+
+#### destroy
+
+Uninstall Helm release.
+
+#### reset
+
+Not implemented.
+
+#### clean
+
+Not implemented.
+
+## KafkaApp
+
+Sub class of [_KubernetesApp_](#kubernetesapp).
```

Comment:
Please also define the `KafkaApp`.

Reply:
What do you mean?

Reply:
IMO there is a definition of KafkaApp missing. The sentence `Sub class of [_KubernetesApp_](#kubernetesapp).` is not a suitable definition for the KafkaApp. 
It represents a Kafka Streams App and a Producer App, both from Streams bootstrap.
In general we should rephrase this. Something like:
```
A KafkaApp is a KubernetesApp. The KafkaApp represents a...
```

Reply:
This is covered under "KafkaApp/Usage" and in "Kafka Streams App" and "Producer App"

I'd prefer to stay consistent with the rest of the document and not have a description outside of "Usage", but if you think that information is missing or hard to understand, I will consider restructuring and/or rewriting parts.